File: rustprojects/primenumber.rs
use std::io;

fn main() {
    println!("Enter a number");

    let mut input = String::new();

    io::stdin()
        .read_line(&mut input)
        .expect("Failed to take input");

    let n: i32 = input.trim().parse().expect("Not a number");

    println!("Number is: {}", n);

    let mut totalfactors = 0;

    for i in 1..n {
        if n%i == 0 {
            totalfactors += 1;
        }
    }

    if totalfactors == 2 {
        println!("prime number");
    } else {
        println!("not a prime number");
    }
}

File: cppprojects/primenumber.cpp
#include <iostream>

bool isprime(int n) {
    int totalfactors = 0;

    for (int i = 1; i <= n; i++) {
        if (n % i == 0) {
            totalfactors++;
        }
    }

    return (totalfactors == 2);
}

int main () {
    int number;

    std::cout << "enter a number" << std::endl;
    std::cin >> number;

    if (isprime(number)) {
        std::cout << "prime number" << std::endl;
    } else {
        std::cout << "not a prime number" << std::endl;
    }

    return 0;
}

